runtime/src/main/codestarts/quarkus/quakus-hilla-maven-codestart/base/pom.xml.tpl.qute

```diff
@@ -0,0 +1,94 @@
+<project>
+{#if hillaVersion }
+    <properties>
+        <hilla.version>{hillaVersion}</hilla.version>
+    </properties>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>dev.hilla</groupId>
+                <artifactId>hilla-maven-plugin</artifactId>
+                <version>$\{hilla.version\}</version>
+                <executions>
+                  <execution>
+                    <goals>
+                      <goal>prepare-frontend</goal>
+                    </goals>
+                  </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+
+    <profiles>
+        <profile>
+            <!-- Hilla Production mode is activated using -Pproduction -->
+            <id>production</id>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>dev.hilla</groupId>
+                        <artifactId>hilla-maven-plugin</artifactId>
+                        <version>$\{hilla.version\}</version>
+                        <executions>
+                            <execution>
+                                <goals>
+                                    <goal>build-frontend</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
+
+    {#if !hillaVersion.matches("^\\d+\\.\\d+\.\\d+$") }
```

Comment:
This would allow any strange version input like: alpha-beta-gamma, is there any possibility to break the generation if an invalid version is pasted?

Reply:
Also in regard of the main section above. Or do we rely on the intelligence of the user?

Reply:
This is expected to be a real Hilla version, so it may only be `N.N.N` or `N.N.N.(alpha|beta|rc)N` or  `N.N-SNAPSHOT`.
In the last two cases, the artifacts should be fetched from Vaadin maven repository